karma.conf.js

```diff
@@ -1,70 +1,84 @@
 // Karma configuration
 // Generated on Sun Jan 20 2019 23:06:22 GMT-0600 (CST)
 
+const isCI = !!process.env.CI;
+const watch = !!process.argv.find(arg => arg.includes('watch')) && !isCI;
+const coverage = !!process.argv.find(arg => arg.includes('--coverage'));
+
 module.exports = function(config) {
   config.set({
-
     // base path that will be used to resolve all patterns (eg. files, exclude)
     basePath: '',
 
+    plugins: [
+      require('karma-jasmine'),
+      require('karma-chrome-launcher'),
+      require('karma-firefox-launcher'),
+      require('karma-safari-launcher'),
+      require('karma-coverage-istanbul-reporter'),
+      require('karma-detect-browsers'),
+      require('@open-wc/karma-esm'),
+    ],
 
     // frameworks to use
     // available frameworks: https://npmjs.org/browse/keyword/karma-adapter
-    frameworks: ['jasmine'],
-
+    frameworks: ['jasmine', 'esm', 'detectBrowsers'],
 
     // list of files / patterns to load in the browser
-    files: [
-      { pattern: './adoptedStyleSheets.js', include: true },
-      'test/*.test.js'
-    ],
-
+    files: [{pattern: 'test/polyfill.test.js', type: 'module', watch: false}],
 
     // list of files / patterns to exclude
-    exclude: [
-    ],
-
+    exclude: [],
 
     // preprocess matching files before serving them to the browser
     // available preprocessors: https://npmjs.org/browse/keyword/karma-preprocessor
-    preprocessors: {
-    },
-
+    preprocessors: {},
 
     // test results reporter to use
     // possible values: 'dots', 'progress'
     // available reporters: https://npmjs.org/browse/keyword/karma-reporter
-    reporters: ['progress'],
-
+    reporters: coverage ? ['progress', 'coverage-istanbul'] : ['progress'],
 
     // web server port
     port: 9876,
 
-
     // enable / disable colors in the output (reporters and logs)
     colors: true,
 
-
     // level of logging
     // possible values: config.LOG_DISABLE || config.LOG_ERROR || config.LOG_WARN || config.LOG_INFO || config.LOG_DEBUG
     logLevel: config.LOG_INFO,
 
-
     // enable / disable watching file and executing tests whenever any file changes
-    autoWatch: true,
+    autoWatch: watch,
 
+    esm: {
+      coverage,
+      compatibility: 'none',
+      nodeResolve: true,
+    },
 
-    // start these browsers
-    // available browser launchers: https://npmjs.org/browse/keyword/karma-launcher
-    browsers: ['Chrome', 'Firefox'],
+    coverageIstanbulReporter: {
+      reports: ['html', 'lcovonly', 'text-summary'],
+      dir: '.coverage',
+      combineBrowserReports: true,
+      skipFilesWithNoCoverage: false,
+    },
 
+    detectBrowsers: {
+      usePhantomJS: false,
+      preferHeadless: true,
+      postDetection(availableBrowsers) {
+        return availableBrowsers.filter(browser => browser !== 'IE');
```

Comment:
There are settings for `detectBrowsers` plugin that detects and starts browsers if they exist and the proper launcher is installed. I do not have `karma-ie-launcher` installed, so it should not run IE, but it does for some reason. So I had to exclude it. But by default it should contain all the browsers you have on your machine.

Reply:
Regarding the Safari Technology Preview: I mentioned it earlier that the original Safari launcher for Karma does not work properly with Safari > 12. I replaced it with `karma-safarinative-launcher`, but I'm not sure it can run the TP.